portal/server/app/route.ts

```diff
@@ -5,6 +5,29 @@ import { getDomain, getSubdomainAndPath } from "@lib/domain_parsing";
 import { redirectToAggregatorUrlResponse, redirectToPortalURLResponse } from "@lib/redirects";
 import { getBlobIdLink, getObjectIdLink } from "@lib/links";
 import { resolveAndFetchPage } from "@lib/page_fetching";
+import logger from "@lib/logger";
+import * as Sentry from "@sentry/node";
+
+function addLoggingArgsToSentry(args: { [key: string]: any }) {
+    Object.entries(args).forEach(([key, value]) => {
+        if (key !== "message") { // Skipping the 'message' key
+            console.log(`${key}: ${value}`)
+            Sentry.setTag(key, value);
+        }
+    });
+}
+logger.setErrorPredicate(args => {
+    addLoggingArgsToSentry(args);
+    Sentry.captureException(new Error(args.message ))
+});
+logger.setWarnPredicate(args => {
+    addLoggingArgsToSentry(args);
+    Sentry.addBreadcrumb({ message: args.message, data: args })
+} );
+logger.setInfoPredicate(args => {
+    addLoggingArgsToSentry(args);
+    Sentry.addBreadcrumb({ message: args.message, data: args })
+} );
```

Comment:
This is the part that the logger predicates are defined for the server portal, so that it utilizes Sentry. 
The `breadcrumbs` are additional information about a trace (i.e. custom logs). 
The `tags` are added to each event so we can query the logs on the Sentry UI.

Reply:
question: isn't the information duplicated if the  tags and the breadcrumbs both contain `args`?  is this useful?